assets/src/edit-story/components/panels/link/index.js

```diff
@@ -112,10 +112,14 @@ function LinkPanel({ selectedElements, pushUpdateForObject }) {
     ({ url, title, icon }) =>
       pushUpdateForObject(
         'link',
-        (prev) => ({
-          desc: title ? title : prev.desc,
-          icon: icon ? toAbsoluteUrl(url, icon) : prev.icon,
-        }),
+        (prev) =>
+          url
+            ? {
+                url,
```

Comment:
Ensures the related url is in sync with the metadata being saved -- perhaps that could improve the issue of metadata being saved together with an empty URL as it happened with the bug.